core/src/test/java/org/apache/iceberg/data/DataTest.java

```diff
@@ -129,24 +135,37 @@ protected boolean supportsVariant() {
     return false;
   }
 
+  protected boolean supportsGeospatial() {
+    return false;
+  }
+
   @ParameterizedTest
   @FieldSource("SIMPLE_TYPES")
   public void testTypeSchema(Type type) throws IOException {
-    Assumptions.assumeThat(
-            supportsUnknown()
-                || TypeUtil.find(type, t -> t.typeId() == Type.TypeID.UNKNOWN) == null)
-        .as("unknown is not yet implemented")
-        .isTrue();
-    Assumptions.assumeThat(
-            supportsTimestampNanos()
-                || TypeUtil.find(type, t -> t.typeId() == Type.TypeID.TIMESTAMP_NANO) == null)
-        .as("timestamp_ns is not yet implemented")
-        .isTrue();
-    Assumptions.assumeThat(
-            supportsVariant()
-                || TypeUtil.find(type, t -> t.typeId() == Type.TypeID.VARIANT) == null)
-        .as("variant is not yet implemented")
-        .isTrue();
+    if (!supportsUnknown()) {
```

Comment:
Can you revert the changes to the other assumptions? I don't think that we need to modify those lines.

Reply:
Reverted